src/endpoint/devices.ts

```diff
@@ -5,24 +5,62 @@ import { Links, Status, SuccessStatusValue } from '../types'
 import { PresentationDevicePresentation } from './presentation'
 
 
-const HEADER_OVERRIDES = {Accept: 'application/vnd.smartthings+json;v=20170916'}
+const HEADER_OVERRIDES = { Accept: 'application/vnd.smartthings+json;v=20170916' }
 
 export interface CapabilityReference {
 	id: string
 	version?: number
 	status?: CapabilityStatus
 }
 
+export interface DeviceProfileReference {
+	id?: string
+}
+
+// Included for backwards compatibility (renamed to match API docs).
+// For new code, use DeviceProfileReference.
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface ProfileIdentifier extends DeviceProfileReference {}
+
+export interface DeviceCategory {
+	name: string
+	categoryType: 'manufacturer' | 'user'
+}
+
+export interface Restrictions {
+	tier: number // integer
+	historyRetentionTTLDays?: number // integer
+	visibleWhenRestricted?: boolean // default: false
+}
+
 export interface Component {
-	id?: string // <^[-_!.~'()*0-9a-zA-Z]{1,36}$>
-	label?: string
+	id: string // <^[-_!.~'()*0-9a-zA-Z]{1,36}$>
 	capabilities: CapabilityReference[]
+	categories: DeviceCategory[]
+	label?: string
+	restrictions?: Restrictions
+	icon?: string
 }
 
 export interface AppDeviceDetails {
 	installedAppId?: string // <^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$>
 	externalId?: string // <= 64 characters
-	profileId?: string
+	profile?: DeviceProfileReference
+}
+
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface BleDeviceDetails {
+	// This object currently contains no properties.
```

Comment:
Is this still necessary with the eslint toggle?

Reply:
The intent of the comment was to describe why it was empty here but now when I reread it, I realize now that it doesn't read that way at all. I was trying to say that the API docs define this object with not properties. I've changed the wording here to "The API defines this object without properties."